nevergrad/benchmark/experiments_gym.py

```diff
@@ -0,0 +1,295 @@
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import typing as tp
+from nevergrad.functions.gym import GymMulti
+from nevergrad.functions.gym import CompilerGym
+from .xpbase import registry
+from .xpbase import create_seed_generator
+from .xpbase import Experiment
+from .optgroups import get_optimizers
+
+
+@registry.register
+def ng_full_gym(
+    seed: tp.Optional[int] = None,
+    randomized: bool = True,
+    multi: bool = False,
+    big: bool = False,
+    memory: bool = False,
+    ng_gym: bool = False,
+    conformant: bool = False,
+) -> tp.Iterator[Experiment]:
+    """Gym simulator. Maximize reward.  Many distinct problems.
+
+    Parameters:
+        seed: int
+           random seed.
+        randomized: bool
+           whether we keep the problem's stochasticity
+        multi: bool
+           do we have one neural net per time step
+        big: bool
+           do we consider big budgets
+        memory: bool
+           do we use recurrent nets
+        ng_gym: bool
+           do we restrict to ng-gym
+        conformant: bool
+           do we restrict to conformant planning, i.e. deterministic controls.
+    """
+    env_names = GymMulti.env_names
+    if ng_gym:
+        env_names = GymMulti.ng_gym
+    seedg = create_seed_generator(seed)
+    optims = ["DiagonalCMA", "OnePlusOne", "PSO", "DiscreteOnePlusOne", "DE", "CMandAS2"]  # type: ignore
+    if multi:
+        controls = ["multi_neural"]
+    else:
+        controls = (
+            [
+                "neural",
+                "structured_neural",
+                # "memory_neural",
+                "stackingmemory_neural",
+                "deep_neural",
+                "semideep_neural",
+                # "noisy_neural",
+                # "noisy_scrambled_neural",
+                # "scrambled_neural",
+                # "linear",
+            ]
+            if not big
+            else ["neural"]
+        )
+    if memory:
+        controls = ["stackingmemory_neural", "deep_stackingmemory_neural", "semideep_stackingmemory_neural"]
+        controls += ["memory_neural", "deep_memory_neural", "semideep_memory_neural"]
+        controls += [
+            "extrapolatestackingmemory_neural",
+            "deep_extrapolatestackingmemory_neural",
+            "semideep_extrapolatestackingmemory_neural",
+        ]
+        assert not multi
+    if conformant:
+        controls = ["stochastic_conformant"]
+    for control in controls:
+        neural_factors = [None] if (conformant or control == "linear") else ([1] if "memory" in control else [3 if big else 1])  # type: ignore
+        for neural_factor in neural_factors:
+            for name in env_names:
+                try:
+                    func = GymMulti(name, control=control, neural_factor=neural_factor, randomized=randomized)
+                except MemoryError:
+                    pass
```

Comment:
This is dangerous because then it falls back to the previous function, you want `continue` instead

Reply:
Agree yes.